hrjobcontract/js/src/job-contract/controllers/revision-list.js

```diff
@@ -29,32 +29,18 @@ define([
         recordedBy: true
       };
 
+      function init(){
+        if(!$scope.revisionDataList){
+          $scope.$broadcast('hrjc-loader-show');
 
-      $scope.createPage = function() {
-        var start = (($scope.currentPage - 1) * $scope.itemsPerPage),
-          end = start + $scope.itemsPerPage;
-
-        $scope.revisionDataListPage = revisionDataListLocal.slice(start, end);
-      }
-
-      $scope.sortBy = function(sortCol, sortReverse) {
-
-        if (typeof sortCol !== 'undefined') {
-
-          if ($scope.sortCol == sortCol) {
-            $scope.sortReverse = !$scope.sortReverse;
-          } else {
-            $scope.sortCol = sortCol;
-          }
-
-        }
-
-        if (typeof sortReverse !== 'undefined') {
-          $scope.sortReverse = sortReverse;
+          // Fetching revision list form ContractRevisionList service
+          ContractRevisionList.fetchRevisions(contractId).then(function(result){
+            Array.prototype.push.apply($scope.revisionList, result.revisionList);
```

Comment:
$scope.revisionList is  an array, so `$scope.revisionList = result.revisionList`.  Array.prototype.push is for objects.